### PR TITLE
ci(workflow): expand integration test groups

### PR DESCRIPTION
**Pull request by kwonoj**
closes WEB-545, WEB-571.

This PR mimic upstream next.js test group config to expand turbopack's integration integration to match with it. To reduce some of duplication, this PR also sets up reusable workflow to setup  / build next.js and share its build artifact via github cache. Since those build artifacts are one-off usage per each test, it doesn't use upload/download artifact which is too slow & heavyweight for those purpose.

### CODE REVIEW

**Comment by kwonoj** [comment]
This may or may not fix WEB-544 as well. I need to dig further once this PR lands.

File: .github/actions/next-integration-stat/src/index.ts [hunk 1]
```diff
@@ -136,35 +177,46 @@ function collectFailedTestResults(
...
-          name: failedTest,
-          data: JSON.parse(testData),
-        };
-      } catch (_) {
-        console.log(`Failed to parse test data`);
-        return null;
+      let failedSplitLogs = logs.split(`failed to pass within`);
+      const ret = [];
+
+      while (!!failedSplitLogs && failedSplitLogs.length >= 1) {
```
Comment (wbinnssmith) [inline .github/actions/next-integration-stat/src/index.ts:183]: Can this ever be falsey?

Comment (kwonoj) [reply .github/actions/next-integration-stat/src/index.ts:183]: yes we consume it here https://github.com/vercel/turbo/pull/3659/files#diff-0e0c2f9cb52224d172853cfbed972f9ac25436bf53078850cd7b3a7168fef0e7R184 so eventually it'll drain

File: .github/actions/next-integration-stat/src/index.ts [hunk 2]
```diff
@@ -566,13 +654,13 @@ function getTestSummary(
 
   if (fixedTests.length > 0) {
     ret += `\n:white_check_mark: **Fixed tests:**\n\n${fixedTests
-      .map((t) => `\t- ${t}`)
+      .map((t) => (t.length > 5 ? `\t- ${t}` : t))
```
Comment (wbinnssmith) [inline .github/actions/next-integration-stat/src/index.ts:657]: Maybe a comment explaining why 5 is significant?

Comment (kwonoj) [reply .github/actions/next-integration-stat/src/index.ts:657]: it's actually no significant random number 😅 I just picked small enough to ensure it have corresponding filename in the logs.

**Review by wbinnssmith — APPROVED** [review]

**Pull request merged**